package.json

```diff
@@ -125,6 +125,166 @@
+                                        "number",
+                                        "boolean"
+                                    ]
+                                },
+                                "type": "object"
+                            }
+                        },
+                        "additionalProperties": false,
+                        "properties": {
+                            "aws": {
+                                "title": "AWS Connection",
+                                "description": "%AWS.configuration.description.awssam.debug.aws%",
+                                "properties": {
+                                    "credentials": {
+                                        "description": "%AWS.configuration.description.awssam.debug.credentials%",
+                                        "type": "string"
+                                    },
+                                    "region": {
+                                        "description": "%AWS.configuration.description.awssam.debug.region%",
+                                        "type": "string"
+                                    }
+                                },
+                                "additionalProperties": false,
+                                "type": "object"
+                            },
+                            "invokeTarget": {
+                                "title": "Invoke Target Properties",
+                                "description": "%AWS.configuration.description.awssam.debug.invokeTarget%",
+                                "properties": {
+                                    "lambdaHandler": {
+                                        "description": "%AWS.configuration.description.awssam.debug.lambdaHandler%",
+                                        "type": "string"
+                                    },
+                                    "projectRoot": {
+                                        "description": "%AWS.configuration.description.awssam.debug.projectRoot%",
+                                        "type": "string"
+                                    },
+                                    "samTemplatePath": {
+                                        "description": "%AWS.configuration.description.awssam.debug.samTemplatePath%",
+                                        "type": "string"
+                                    },
+                                    "samTemplateResource": {
+                                        "description": "%AWS.configuration.description.awssam.debug.samTemplateResource%",
+                                        "type": "string"
+                                    },
+                                    "target": {
+                                        "description": "%AWS.configuration.description.awssam.debug.target%",
+                                        "type": "string",
+                                        "enum": [
```

Comment:
I suspect if you make factor out the enum into it's own definition, it will generate a proper typescript enum instead of a union type.